### ISSUE TITLE
`ipython notebook` is going away...

### ISSUE THREAD

**Issue by arokem** [thunder-project/thunder#254]
You've probably already seen this one:

```
[TerminalIPythonApp] WARNING | Subcommand `ipython notebook` is deprecated and will be removed in future versions.
[TerminalIPythonApp] WARNING | You likely want to use `jupyter notebook`... continue in 5 sec. Press Ctrl-C to quit now.
```

Is this something that needs to change upstream here? https://github.com/apache/spark/blob/master/bin/pyspark

**Comment by bpirvu** [comment]
Changing line 52 from 

`PYSPARK_DRIVER_PYTHON="ipython"`

to

`PYSPARK_DRIVER_PYTHON="jupyter"`

works for me to solve this issue, but I'm not sure if it's sustainable as I'm not familiar with the rest of the code...

**Comment by freeman-lab** [comment]
Thanks both! In the imminent 1.0.0 release there will no longer be a `thunder-ec2`, so this won't strictly show up anymore, but I'd like to describe the EC2 launching workflow in a blog post or something, and when I do I'll definitely incorporate this change!

**Comment by arokem** [comment]
Cheers! I've been using `thunder-ec2` on its own as a way to get a spark
cluster going -- it's super useful! Any chance it will continue to be
maintained somewhere else?

**Comment by freeman-lab** [comment]
Oh that's cool! Yes, definitely want to, in some form =) The main difficulty was that wrapping `spark-ec2` as we were doing became a real maintenance burden, because `spark-ec2` itself changed frequently and was basically one giant, very complicated script.

Other than installing Thunder, the main thing `thunder-ec2` did beyond `spark-ec2` was install more recent versions of scientific Python, and give what I considered a friendlier CLI experience =)

We could easily reimplement those goals in a new standalone CLI. But I think I'd rather base future work off of [`flintrock`](https://github.com/nchammas/flintrock) by @nchammas which is *way* faster, and should be more composable.

Would be very curious to get your thoughts on `spark-ec2` vs `thunder-ec2` vs `flintrock` (if you try it), and maybe team up on a new thing?

Also, if your curious, some of this is discussed on the new WIP doc site (http://docs.thunder-project.org/), obviously it's still incomplete, but feedback welcome!

**Comment by nchammas** [comment]
Hi @freeman-lab! Yes, one of the design goals of Flintrock is to be [composable](https://github.com/nchammas/flintrock#automated-pipelines).

Though I have no current plans to expose a Python API, Flintrock aims to be a well-behaved command-line application. Exit codes are deliberately chosen and match Unix conventions, and output is directed to `stdout` and `stderr` appropriately.

In the case where you might want to consume data from Flintrock -- typically from `flintrock describe` -- the output comes to you in YAML, so it should be easy to parse and use.

If y'all run into any issues using Flintrock as part of the Thunder Project, definitely let me know. I'm interested in supporting y'all if you think our projects are a fit.

> The main difficulty was that wrapping spark-ec2 as we were doing became a real maintenance burden, because spark-ec2 itself changed frequently and was basically one giant, very complicated script.

One caveat here I should highlight is that Flintrock is still relatively young (we released 0.4 a couple of weeks ago) and its interface may also change in backwards incompatible ways. I don't expect that to happen, but until 1.0 obviously I can't make any hard promises.
